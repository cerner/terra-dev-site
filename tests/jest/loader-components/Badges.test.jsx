import React from 'react';
import Badge from '../../../src/loader-components/_Badges';

const packageName = '@cerner/terra-dev-site';
const packageUrl = 'https://engineering.cerner.com/terra-dev-site';
const packageVersion = '2.13.0';
const packageSrcCode = 'https://github.com/cerner/terra-dev-site';

describe('Badges', () => {
  it('should render only custom package badge when url and version props are set', () => {
    const wrapper = shallow(<Badge
      name={packageName}
      url={packageUrl}
      version={packageVersion}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render only npm badge when url is not specified', () => {
    const wrapper = shallow(<Badge
      name={packageName}
      version={packageVersion}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render npm badge and source code badge when url is not specified and repository url is specified', () => {
    const wrapper = shallow(<Badge
      src={packageSrcCode}
      name={packageName}
      version={packageVersion}
    />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render custom package badge and source code badge when both url and repository url is specified', () => {
    const wrapper = shallow(<Badge
      src={packageSrcCode}
      name={packageName}
      url={packageUrl}
      version={packageVersion}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
