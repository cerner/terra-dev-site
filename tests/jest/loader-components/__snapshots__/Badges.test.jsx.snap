// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badges should render custom package badge and source code badge when both url and repository url is specified 1`] = `
<div
  className="badge-container"
>
  <a
    className="badge"
    href="https://engineering.cerner.com/terra-dev-site"
  >
    <span
      className="badge-name"
    >
      package
    </span>
    <span
      className="badge-version"
    >
      v2.13.0
    </span>
  </a>
  <a
    className="badge"
    href="https://github.com/cerner/terra-dev-site"
  >
    <span
      className="badge-name"
    >
      github
    </span>
    <span
      className="badge-version"
    >
      source
    </span>
  </a>
</div>
`;

exports[`Badges should render npm badge and source code badge when url is not specified and repository url is specified 1`] = `
<div
  className="badge-container"
>
  <a
    className="badge"
    href="https://www.npmjs.org/package/@cerner/terra-dev-site/v/2.13.0"
  >
    <span
      className="badge-name"
    >
      npm
    </span>
    <span
      className="badge-version"
    >
      v2.13.0
    </span>
  </a>
  <a
    className="badge"
    href="https://github.com/cerner/terra-dev-site"
  >
    <span
      className="badge-name"
    >
      github
    </span>
    <span
      className="badge-version"
    >
      source
    </span>
  </a>
</div>
`;

exports[`Badges should render only custom package badge when url and version props are set 1`] = `
<div
  className="badge-container"
>
  <a
    className="badge"
    href="https://engineering.cerner.com/terra-dev-site"
  >
    <span
      className="badge-name"
    >
      package
    </span>
    <span
      className="badge-version"
    >
      v2.13.0
    </span>
  </a>
</div>
`;

exports[`Badges should render only npm badge when url is not specified 1`] = `
<div
  className="badge-container"
>
  <a
    className="badge"
    href="https://www.npmjs.org/package/@cerner/terra-dev-site/v/2.13.0"
  >
    <span
      className="badge-name"
    >
      npm
    </span>
    <span
      className="badge-version"
    >
      v2.13.0
    </span>
  </a>
</div>
`;
